snapcraft/internal/common.py

```diff
@@ -149,10 +149,16 @@ def reset_env():
 
 def link_or_copy(source, destination, follow_symlinks=False):
     # Hard-link this file to the source. It's possible for this to
-    # fail (e.g. a cross-device link), so as a backup plan we'll
-    # just copy it.
+    # fail (e.g. a cross-device link, or permission denied), so as a backup
+    # plan we'll just copy it.
     try:
-        os.link(source, destination, follow_symlinks=follow_symlinks)
+        # Note that follow_symlinks doesn't seem to work for os.link, so we'll
+        # implement this logic ourselves using realpath.
+        source_path = source
+        if follow_symlinks:
+            source_path = os.path.realpath(source)
+
+        os.link(source_path, destination, follow_symlinks=False)
```

Comment:
Yeah but it defaults to True and I want to make sure this continues working even if the bug is fixed upstream.